### PR TITLE
Getter for 'from', 'to' event's configuration

### PR DESCRIPTION
**Pull request by lucatacconi**
| Q             | A
| ------------- | ---
| Fixed issue | #59

Inside the Event class there are several methods that allow you to have details on the individual task such as getId(), getSummaryForDisplay(), getExpression(), etc..

However, by setting the task lifetime with the beetween(), from() or to() methods there is no way to read this configuration from the class.

I added two protected variables that allow saving the from and to information, which will only be used for displaying the information and will not influence the behavior of the Event class in any way. I then added the two methods getFrom() and getTo() to display the information.

### CODE REVIEW

File: src/Event.php [hunk 1]
```diff
@@ -88,6 +88,20 @@ class Event implements PingableInterface
      */
     protected $timezone;
 
+    /**
+     * Datetime or time since the task is evaluated and possibly executed only for display purposes.
+     *
+     * @var \DateTime|string|null
+     */
+    protected $disply_from = null;
```
Comment (PabloKowalczyk) [inline src/Event.php]: No need to add `display` prefix, in both cases.

Comment (PabloKowalczyk) [inline src/Event.php]: Use union type instead of `@var`, in both cases.

File: src/Event.php [hunk 2]
```diff
@@ -921,6 +939,26 @@ public function getExpression()
         return $this->expression;
     }
 
+    /**
+     * Get the 'from' configuration for the event if present.
+     *
+     * @return \DateTime|string|null
+     */
+    public function getFrom()
```
Comment (PabloKowalczyk) [inline src/Event.php]: Use union type for return, in both cases.

File: src/Event.php [hunk 3]
```diff
@@ -921,6 +939,26 @@ public function getExpression()
...
+    {
+        return $this->disply_from;
+    }
+
+    /**
+     * Get the 'to' configuration for the event if present.
+     *
+     * @return \DateTime|string|null
+     */
+    public function getTo()
```
Comment (PabloKowalczyk) [inline src/Event.php]: Both methods need unit tests.

Comment (lucatacconi) [reply src/Event.php]: Done

File: src/Event.php (hunk 1 above)
Comment (lucatacconi) [reply src/Event.php]: Done

File: src/Event.php [hunk 4]
```diff
@@ -88,6 +88,20 @@ class Event implements PingableInterface
      */
     protected $timezone;
 
+    /**
+     * Datetime or time since the task is evaluated and possibly executed only for display purposes.
+     *
+     * @var \DateTime|string|null
+     */
+    protected $from = null;
```
Comment (PabloKowalczyk) [inline src/Event.php]: Add union type (`protected \DateTime|string|null $from = null;`) in both cases.

File: src/Event.php (hunk 2 above)
Comment (PabloKowalczyk) [reply src/Event.php]: Bump.

File: tests/Unit/EventTest.php [hunk 5]
```diff
@@ -154,6 +154,35 @@ public function test_cron_life_time(): void
         );
     }
 
+    public function test_cron_life_time_getters(): void
```
Comment (PabloKowalczyk) [inline tests/Unit/EventTest.php]: Split this test into two tests, one per method.

**Comment by PabloKowalczyk** [comment]
CI reported some issues https://github.com/crunzphp/crunz/actions/runs/6561933443/job/17861565426?pr=68

File: tests/Unit/EventTest.php (hunk 5 above)
Comment (lucatacconi) [reply tests/Unit/EventTest.php]: Yes sir :)

File: src/Event.php (hunk 1 above)
Comment (lucatacconi) [reply src/Event.php]: Done

File: src/Event.php (hunk 2 above)
Comment (lucatacconi) [reply src/Event.php]: Done

File: src/Event.php (hunk 4 above)
Comment (lucatacconi) [reply src/Event.php]: Done

File: tests/Unit/EventTest.php (hunk 5 above)
Comment (lucatacconi) [reply tests/Unit/EventTest.php]: Done

File: tests/Unit/EventTest.php [hunk 6]
```diff
@@ -154,6 +154,50 @@ public function test_cron_life_time(): void
         );
     }
 
+    public function test_cron_life_time_get_from(): void
```
Comment (PabloKowalczyk) [inline tests/Unit/EventTest.php]: Almost there :)
- name test cases like `test_get_from` and `test_get_to`
- test `between` in other test case
- use data provider (can be one for both cases) to test `string` and `\DateTime` in `from`/`to`
- use camelCase for variables' names
- use `assertSame` in cases like `self::assertTrue($date_from === $event->getFrom());`

File: tests/Unit/EventTest.php [hunk 7]
```diff
@@ -154,6 +154,45 @@ public function test_cron_life_time(): void
...
+
+        $event = new Event($this->id, 'php foo');
+        $event->from($dateFrom);
+
+        self::assertSame($dateFrom, $event->getFrom());
+    }
+
+    public function test_get_to(): void
+    {
+        $timezone = new \DateTimeZone('UTC');
```
Comment (PabloKowalczyk) [inline tests/Unit/EventTest.php]: Looks like `$timezone` is not used in any test, should be removed.

File: tests/Unit/EventTest.php [hunk 8]
```diff
@@ -154,6 +154,45 @@ public function test_cron_life_time(): void
         );
     }
 
+    public function test_get_from(): void
```
Comment (PabloKowalczyk) [inline tests/Unit/EventTest.php]: What about "use data provider (can be one for both cases) to test `string` and `\DateTime` in from/to"?

File: tests/Unit/EventTest.php (hunk 6 above)
Comment (lucatacconi) [reply tests/Unit/EventTest.php]: Test cases renamed.
Test case between introduced.
Variables' name renamed in camelCase.

Do you prefer to have the dataproviders all together at the end of the controls file near the other one or near the functions that use them?

File: tests/Unit/EventTest.php (hunk 8 above)
Comment (lucatacconi) [reply tests/Unit/EventTest.php]: In progress.

File: tests/Unit/EventTest.php (hunk 6 above)
Comment (PabloKowalczyk) [reply tests/Unit/EventTest.php]: > Do you prefer to have the dataproviders all together at the end of the controls file near the other one or near the functions that use them?

End of file, before private methods, please.

**Comment by lucatacconi** [comment]
All jobs done

**Comment by lucatacconi** [comment]
Are there any other changes you think are needed?

**Comment by PabloKowalczyk** [comment]
Thanks @lucatacconi

**Pull request merged**